vep/__init__.py

```diff
@@ -321,17 +325,35 @@ def install_pip_requirements(self, pip):
             pip_args += ['--cache-dir', self.pip_cache]
         pip(_out=print_line, *pip_args)
 
+    def which(self, program):
+        def is_exe(fpath):
+            return os.path.isfile(fpath) and os.access(fpath, os.X_OK)
+
+        fpath, fname = os.path.split(program)
+        if fpath:
+            if is_exe(program):
+                return program
+        else:
+            for path in os.environ["PATH"].split(os.pathsep):
+                exe_file = os.path.join(path, program)
+                if is_exe(exe_file):
+                    return exe_file
+
+        return None
+
     def create_virtualenv(self):
         rm = sh.Command('rm')
         print("deleting previous virtual environment")
         rm('-f', '-r', self.target)
         print("creating new virtual environment")
-        # we've got to use the virtualenv that is in ve-packager, not the system virtualenv,
-        # which might use a different version of python.
-        virtualenv = sh.Command('%s/virtualenv' % os.path.dirname(sys.executable))
+        # Find virtualenv that is in the system path, if there are multiple use the 1st one found
+        # Python2 uses virtualenv provided by deb packages - /usr/bin/virtualenv
+        # Python3 will use virtualenv provided by Pip - /usr/local/bin/virtualenv
+        virtualenv = sh.Command(self.which('virtualenv'))
         # check if there is a virtualenv alongside whatever python we are using, and use that
         if os.path.exists('%s/virtualenv' % os.path.dirname(self.python)):
             virtualenv = sh.Command('%s/virtualenv' % os.path.dirname(self.python))
```

Comment:
If you build a package using `/usr/bin/python3`, and the Debian package `virtualenv` is installed, it looks to me like this will override the `/usr/bin`/`usr/local/bin` logic above. Is that what you want?

Reply:
You are correct, but I had to leave that there for Python2 compat.

In Python3 land the container will be built using the upstream pypi pip and the virtualenv deb package should never be installed.  As we go through and touch the various repos it'll be something to look out for.